### PR TITLE
[SPARK-31906][SQL][DOCS] Enhance comments in NamedExpression.qualifier

### PR DESCRIPTION
**Pull request by zhulipeng**
### Why are the changes needed?
The qualifier name should contains catalog name.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
UT.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
Seems fine, cc @cloud-fan FYI

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala [hunk 1]
```diff
@@ -85,6 +85,7 @@ trait NamedExpression extends Expression {
    *    e.g. top level attributes aliased in the SELECT clause, or column from a LocalRelation.
    * 2. Seq with a Single element: either the table name or the alias name of the table.
    * 3. Seq with 2 elements: database name and table name
+   * 4. Seq with 3 elements: catalog name, database name and table name
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala:88]: namespace instead?

Comment (lipzhu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala:88]: > namespace instead?

 Sorry for the late response. You mean `catalog namespace`?

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala:88]: I think he meant to document the namespace support in catalogV2, which is related but I think can be done separately.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala (hunk 1 above)
Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala:88]: Yea, it looks fine as it is.

**Review by maropu — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**